conversations/impl/src/test/org/sakaiproject/conversations/impl/ConversationsServiceTests.java

```diff
@@ -1841,6 +1845,140 @@ public void markPostsViewed() {
+        assertEquals(title2, savedBean.title);
+
+        clearInvocations(gradingService);
+
+        // Now let's simulate the user editing the topic and selecing the create a grading item
+        // checkbox. This should mean that the code recognises that we already have an external
+        // grading item for this topic, and it should then just update the existing grading item
+
+        savedBean.graded = true;
+        savedBean.createGradingItem = true;
+        savedBean.title = title1;
+
+        savedBean = saveTopic(savedBean);
+
+        assertEquals(title1, savedBean.title);
+        verify(gradingService).updateExternalAssessment(anyString(), anyString(), anyString(), any(), anyString(), anyDouble(), any(), anyBoolean());
+
+        Long internalGradingItemId = 231L;
+
+        // Now let's simulate the user picking another existing grading item, ie: deselecting the
+        // previously created external grading item. The previously associated external grading item
+        // should be removed.
+        savedBean.graded = true;
+        savedBean.createGradingItem = false;
+        savedBean.gradingItemId = internalGradingItemId;
+        savedBean.gradingPoints = 33D;
+
+        when(gradingService.getAssignment(site1Id, internalGradingItemId)).thenReturn(ass);
+
+        savedBean = saveTopic(savedBean);
+        verify(gradingService).removeExternalAssignment(site1Id, topicRef);
+        assertEquals(internalGradingItemId, savedBean.gradingItemId);
+
+        // Now lets simulate the user picking another existing grading item, so switching from one
+        // "internal" grading item to another "internal" grading item
+        Long internalGradingItemId2 = 232L;
+        savedBean.graded = true;
+        savedBean.createGradingItem = false;
+        savedBean.gradingItemId = internalGradingItemId2;
+        savedBean.gradingPoints = 43D;
+
+        when(gradingService.getAssignment(site1Id, savedBean.gradingItemId)).thenReturn(ass);
+
+        savedBean = saveTopic(savedBean);
+        assertEquals(internalGradingItemId2, savedBean.gradingItemId);
+
+        clearInvocations(gradingService);
+
+        savedBean.graded = true;
+        savedBean.title = "Hola, Mundo";
+
+        savedBean = saveTopic(savedBean);
+
+        assertEquals("Hola, Mundo", savedBean.title);
+        verify(gradingService).updateExternalAssessment(anyString(), anyString(), anyString(), any(), anyString(), anyDouble(), any(), anyBoolean());
+
+        clearInvocations(gradingService);
+
+        savedBean.graded = false;
+        savedBean = saveTopic(savedBean);
+
+        assertNull(savedBean.gradingItemId);
+    }
+
+    private TopicTransferBean saveTopic(TopicTransferBean topicBean) {
+
+        try {
+            return conversationsService.saveTopic(topicBean, false);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
log.error

Reply:
It's just a unit test. I usually use printStackTrace if it is not an expected exception.